tests/integration/test_remoted/test_multi_groups/test_merged_mg_file_content.py

```diff
@@ -0,0 +1,202 @@
+shared_file_path = os.path.join(shared_folder_path, DEFAULT_TESTING_GROUP_NAME, shared_file_name)
+response_data = None
+elapsed_time = 2
+wait_time = 1
+expected_line = f"!0 {shared_file_name}"
+
+# Configuration
+local_internal_options = {'remoted.shared_reload': f"{wait_time}"}
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+test_cases_path = os.path.join(test_data_path, 'test_cases')
+tcases_data = os.path.join(test_cases_path, 'case_file_actions.yaml')
+configuration_parameters, configuration_metadata, test_case_ids = configuration.get_test_cases_data(tcases_data)
+
+
+# Fixtures
+@pytest.fixture(scope="module")
+def restart_remoted():
+    """Restart the wazuh-remoted daemon."""
+
+    control_service('restart', daemon=REMOTE_DAEMON)
+    check_daemon_status(target_daemon=REMOTE_DAEMON)
+
+    yield
+
+    control_service('stop', daemon=REMOTE_DAEMON)
+    check_daemon_status(target_daemon=REMOTE_DAEMON, running_condition=False)
+
+
+@pytest.fixture(scope='function')
+def prepare_environment(request, register_agent):
+    """Configure a custom environment for testing."""
+
+    new_agent_group()
+
+    agent_id = getattr(request.module, 'response_data')['id']
+
+    # Delete this in 4.4, we have to use the agent_groups script to assign the groups
+    with open(f"{tools.WAZUH_PATH}/queue/agent-groups/{agent_id}", 'w') as agent_group_file:
+        agent_group_file.write(f"default,{DEFAULT_TESTING_GROUP_NAME}")
+
+    yield
+
+    remove_agent_group(DEFAULT_TESTING_GROUP_NAME)
+    delete_file(os.path.join(groups_folder, agent_id))
+
+
+@pytest.fixture(scope='function')
+def register_agent(request):
+    """Register an agent via API."""
+
+    api_details = get_api_details_dict()
+    data = {
+        'name': agent_name,
+        'ip': agent_ip
+    }
+    url = f"{api_details['base_url']}/agents"
+    response = requests.post(url=url, headers=api_details['auth_headers'], json=data, verify=False)
+    response_data = response.json()
+
+    if response.status_code != 200:
+        raise RuntimeError(f"Error registering an agent: {response_data}")
+
+    setattr(request.module, 'response_data', response_data['data'])
+
+    yield
+
+    registered_agent = getattr(request.module, 'response_data')['id']
+    remove_agents(agents_id=[registered_agent], remove_type='api')
+
+
+def manipulate_file(action, file_path):
+    if action == 'create':
+        f = open(file_path, "w")
+        f.close()
+    else:
+        delete_file(file_path)
+
+
+@pytest.mark.parametrize('metadata', configuration_metadata, ids=test_case_ids)
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+def test_merged_mg_file_content(metadata, configure_local_internal_options_module, restart_remoted,
+                                prepare_environment):
+    '''
+    description: Check the content of the merged.mg file that wazuh-remoted compiles for multi-groups.
+
+    wazuh_min_version: 4.2.2
+
```

Comment:
Please, add the left items of the test documentation. You can follow [this schema](https://github.com/wazuh/wazuh-qa/wiki/QA-Documentation---How-to-document-a-test-using-Schema-2.0) to do so.

Then, you must run the `qa-docs` tool to validate the documentation, e.g.:

`qa-docs -p /root/wazuh-qa/tests --types integration --components test_remoted --suites test_multi_groups --modules test_merged_mg_file_content`

[Here](https://github.com/wazuh/wazuh-qa/wiki/QADOCS-tool-use-guide#examples) you have some examples of how to use the tool.

**Note:** `test_phases` is an optional field, but it is desired to include it in the documentation to better explain what the test does.